akka-actor-typed/src/main/scala/akka/actor/typed/internal/routing/RoutingLogic.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.internal.routing
+import akka.actor.typed.ActorRef
+import akka.annotation.InternalApi
+import akka.dispatch.forkjoin.ThreadLocalRandom
+
+/**
+ * Kept in the behavior, not shared between instances so is allowed to be stateful/mutable.
+ *
+ * INTERNAL API
+ */
+@InternalApi
+sealed private[akka] trait RoutingLogic[T] {
+  /**
+   * @param routees available routees, will contain at least one element. Must not be mutated by select logic.
+   */
+  def selectRoutee(routees: Array[ActorRef[T]]): ActorRef[T]
```

Comment:
if we ever make RoutingLogic public API we might not want to pass around arrays, but that is a later concern